opensearch-project/sql
# Minor clean up of datetime and other classes (#198)

Signed-off-by: MitchellGale-BitQuill <mitchellg@bitquilltech.com>
Signed-off-by: Yury-Fridlyand <yury.fridlyand@improving.com>
Co-authored-by: Yury-Fridlyand <yury.fridlyand@improving.com>

### Description
Clean up lexer to make future changes easier. Make changes to documentation and code to make cleaner. 

* Groom ANTLR grammar files to make future changes easier.
* Fix `EqualsAndHashCode` compilation warnings
>
```
Generating equals/hashCode implementation but without a call to superclass, even though this class does not extend java.lang.Object. If this is intentional, add '@EqualsAndHashCode(callSuper=false)' to your type.
```
* Move `ExprTimestampValue.ZONE` constant to `DateTimeUtils`
* Move `ExprDatetimeValue` formatter to `DateTimeFormatters`
* Sort alphabetically functions in `DateTimeFunction` and `MathematicalFunction`
* Replace `FunctionDSL` import by static imports to simplify code
* Remove tests for `NULL` and `MISSING` values
* Add validation for non-zero divider
* Remove `Environment` mock from some unit tests, because it is not needed anymore
* Groom docs to eliminate `rst` validator warnings
* Add doctests for some functions
* Make IT for `now`-like functions truly parametrized
* Rename some PPL grammar entities to be consistent with SQL ones
* Change doctest cluster type to `archive` which has built-in modules - this requires for some added doctests

See team review and discussion in https://github.com/Bit-Quill/opensearch-project-sql/pull/198
 
### Check List

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
